scripts/parse.py

```diff
@@ -170,36 +189,48 @@ def parsesalt(infile, outfile, nestinglevel):
     salt = bytearray2hex(readbytearray(infile, 16))
     outfile.write(salt + ' (salt)\n')
 
-def parsesizedcontent(infile, outfile, nestinglevel, desc):
+def parsesizedcontent(infile, outfile, nestinglevel, enumid):
     length = readshort(infile)
     text = readstring(infile, length)
-    outfile.write('"%s"%s\n' % (text, desc2suffix(desc)))
+    valuedescSet = None
+    # See if text parses to int, it might be a known constant value
+    try:
+        value = int(text, 0)
+        valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
+    except ValueError:
+        pass
+    outfile.write('"%s"%s\n' % (text, desc2suffix(enum2desc(enumid), valuedescSet)))
 
 def parsearrayofenumblockarrays(infile, outfile, nestinglevel):
     size = readshort(infile)
     outfile.write('arrayofenumblockarrays size %d\n' % size)
     for i in range(size):
         parseenumblockarray(infile, outfile, nestinglevel + 1, True, prefix=index2prefix(i))
 
-def parseonebyte(infile, outfile, nestinglevel, desc):
+def parseonebyte(infile, outfile, nestinglevel, enumid):
     value = readbyte(infile)
-    outfile.write('%02X%s\n' % (value, desc2suffix(desc)))
+    valuedescSet = enumwithvalue2valuedesc(knownTAConstants, enumid, value)
```

Comment:
I think we should combine enumwithvalue2valuedesc and enum2desc, because you're giving the enum to the first anyway. That will remove some duplication.

Reply:
Yep, agreed.